tests/test_csc.py

```diff
@@ -74,15 +77,30 @@ async def test_telemetry(self):
                 m_ldco48_temperature=19,
                 m_ldco48_temperature_2=19,
             )
+            with self.subTest("Test fault"):
+                self.csc.simulator.device.do_change_m_cpu800_18_power("FAULT")
+                await self.assert_next_sample(
+                    topic=self.remote.evt_summaryState,
+                    summaryState=salobj.State.FAULT,
+                    flush=True,
+                )
 
     async def test_change_wavelength(self):
         async with self.make_csc(initial_state=salobj.State.ENABLED, simulation_mode=1):
             await self.remote.cmd_changeWavelength.set_start(
                 wavelength=700, timeout=STD_TIMEOUT
             )
+            await self.assert_next_sample(
+                topic=self.remote.evt_wavelengthChanged, wavelength=700
+            )
             await self.assert_next_sample(
                 topic=self.remote.tel_wavelength, wavelength=700, flush=True
             )
+            with self.subTest("Test out of bounds"):
+                with self.assertRaises(salobj.AckError):
```

Comment:
I very gently suggest `with pytest.raises` as being a bit shorter and easier to read.

(I feel much more strongly about the other `self.assertFoo` methods that can be replaced by bare asserts -- personally I find the bare asserts much more readable).